core/pkg/service/flag-evaluation/connect_service.go

```diff
@@ -94,30 +95,39 @@ func (s *ConnectService) setupServer(svcConf service.Configuration) (net.Listene
 	path, handler := schemaConnectV1.NewServiceHandler(fes)
 	mux.Handle(path, handler)
 
-	mdlw := middleware.NewHTTPMetric(middleware.Config{
+	s.server = http.Server{
+		ReadHeaderTimeout: time.Second,
+		Handler:           handler,
+	}
+
+	go bindMetrics(s, svcConf)
+
+	// Add middlewares
+
+	metricsMiddleware := metricsmw.NewHTTPMetric(metricsmw.Config{
 		Service:        "openfeature/flagd",
```

Comment:
Since you are working on this : Service name `"openfeature/flagd"` is exposed through `service.Configuration`. We can improve it here :)